umapi_client/functional.py

```diff
@@ -308,6 +320,48 @@ def delete_account(self):
         self.append(removeFromDomain={})
         return None
 
+    def maybe_split_groups(self, max_groups):
+        """
+        Check if group lists in add/remove directives should be split and split them if needed
+        :param max_groups: Max group list size
+        :return: True if at least one command was split, False if none were split
+        """
+        new_commands = []
+        # return True if we split at least once
+        maybe_split = False
+        for command in self.commands:
+            step_key, step_args = next(six.iteritems(command))
+            if step_key not in [StepKeys.add.name, StepKeys.remove.name, StepKeys.addRoles.name]:
+                continue
```

Comment:
You have to add the current command to new_commands before you continue, or you will lose it.